### ISSUE TITLE
non-normalized tangent vectors 

### ISSUE THREAD

**Issue by HelliceSaouli** [sebastianlipponer/surface_splatting#1]
Hello sir, can you please explain to me how you exactly computed non-normalized tangent vectors u and v

**Comment by sebastianlipponer** [comment]
Are you referring to the computation of u and v for triangles?

https://github.com/sebastianlipponer/surface_splatting/blob/c376d6ce0058d0aa6a3e52ac2270913e0be3e1e4/surface_splatting/main.cpp#L109-L167

**Comment by HelliceSaouli** [comment]
yes the t1, and t2 ? and is it possible to compute it if we are not loading a triangular mesh. ?

**Comment by sebastianlipponer** [comment]
Whats the actual representation of your input geometry?

The code above converts a triangle mesh to a point based representation by computing the circumellipse for each triangle of the triangle mesh. The vectors t1 and t2 are the ellipses semi-major and semi-minor axis. 

While the code allows you to render a triangle mesh using surface splatting in meaningful way, the resulting point based representation is certainly not an optimal reconstruction of the given input geometry.

You may want to use a full-blown point-based surface reconstruction algorithm. For example the technique described by Wu and Kobbelt [1].

[1] [Optimized Sub-Sampling of Point Sets for Surface Splatting](https://www.graphics.rwth-aachen.de/media/papers/splatting-eg041.pdf)
